src/qibolab/instruments/simulator/backends/generic.py

```diff
@@ -0,0 +1,284 @@
+from collections import OrderedDict
+from typing import List, Optional, Union
+
+import numpy as np
+from IPython.display import Latex, display
+
+
+def dec_to_basis_string(
+    x: int, nlevels: list = [2], to_string: bool = True
+) -> Union[str, list]:
+    """Converts an integer to a generalized bitstring in the computation basis
+    of the full Hilbert space.
+
+    Args:
+        x (int): The integer (decimal number) to be converted.
+        nlevels (list): The list of nlevels to convert the decimal number to. Defaults to [2].
+        to_string (bool): Flag to return the result as a string. Defaults to True.
+
+    Returns:
+        str or list: Generalized bitstring of x. If to_string is True, the result is returned as a string, otherwise as a list of integers.
+    """
+
+    nqubits = len(nlevels)
+
+    output_list = []
+    y = x
+    for i in range(nqubits):
+        if i < nqubits - 1:
```

Comment:
```suggestion
    for i in range(nqubits - 1):
```
and then place the `else` outside the `for` loop, just acting on `y` (it already does not depend on `i`).

Reply:
Fixed in 4aa6551 and 7634859